data/csv.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+)
+
+type csvDataSource struct {
+	// Collection is a file path to a CSV file
+	Collection string
+}
+
+func NewCSVDataSource(csvPath string) csvDataSource {
+	return csvDataSource{Collection: csvPath}
+}
+
+func (ds csvDataSource) ReadCollection() [][]string {
```

Comment:
Normally, the receiver must start with the same letter as the struct value. And the shorter the better in this case.
```suggestion
func (c csvDataSource) ReadCollection() [][]string {
```